pkg/npm/tools.go

```diff
@@ -0,0 +1,177 @@
+package npm
+
+import (
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+// Tool encapsulates the commands and configuration for a package manager tool.
+type Tool struct {
+	Name       string
+	ExecRunner ExecRunner
+	InstallCmd []string
+	RunCmd     []string
+	PublishCmd []string
+	AddCmd     []string
+	RemoveCmd  []string
+	TestCmd    []string
+	BuildCmd   []string
+}
+
+var (
+	ToolNPM = Tool{
+		Name:       "npm",
+		InstallCmd: []string{"ci"},
+		RunCmd:     []string{"run"},
+		PublishCmd: []string{"publish"},
+		AddCmd:     []string{"install"},
+		RemoveCmd:  []string{"uninstall"},
+		TestCmd:    []string{"run", "test"},
+		BuildCmd:   []string{"run", "build"},
+	}
+	ToolYarn = Tool{
+		Name:       "yarn",
+		InstallCmd: []string{"install", "--frozen-lockfile"},
+		RunCmd:     []string{"run"},
+		PublishCmd: []string{"publish"},
+		AddCmd:     []string{"add"},
```

Comment:
AddCmd,RemoveCmd,Testcmd and BuildCmd is not serving any purpose atm.
so we can remove it for now and add it back later when there is a requirement.